pandas/core/ops.py

```diff
@@ -752,31 +758,36 @@ def wrapper(left, right, name=name, na_op=na_op):
         na_op = converted.na_op
 
         if isinstance(rvalues, ABCSeries):
-            name = _maybe_match_name(left, rvalues)
             lvalues = getattr(lvalues, 'values', lvalues)
             rvalues = getattr(rvalues, 'values', rvalues)
             # _Op aligns left and right
         else:
-            if isinstance(rvalues, pd.Index):
-                name = _maybe_match_name(left, rvalues)
-            else:
-                name = left.name
             if (hasattr(lvalues, 'values') and
-                    not isinstance(lvalues, pd.DatetimeIndex)):
+                    not isinstance(lvalues, ABCDatetimeIndex)):
                 lvalues = lvalues.values
 
+        res_name = _get_series_op_result_name(left, right)
         result = wrap_results(safe_na_op(lvalues, rvalues))
         return construct_result(
             left,
             result,
             index=left.index,
-            name=name,
+            name=res_name,
             dtype=dtype,
         )
 
     return wrapper
 
 
+def _get_series_op_result_name(left, right):
+    # `left` is always a Series object
```

Comment:
i suggested a simplification

Reply:
> i suggested a simplification

I'm guessing you're referring to [this](https://github.com/pandas-dev/pandas/pull/18884#discussion_r159095783), which I responded to [here](https://github.com/pandas-dev/pandas/pull/18884#discussion_r159097773).

Reply:
ok, rather than creating another function like this, just in-line it as its only used once. .